worlds/lingo/items.py

```diff
@@ -0,0 +1,98 @@
+from typing import Dict, NamedTuple, Optional, List
+from BaseClasses import Item, MultiWorld
+from .Options import get_option_value
+from .static_logic import StaticLingoLogic
+
+
+class ItemData(NamedTuple):
+    """
+    ItemData for an item in Lingo
+    """
+    code: Optional[int]
+    progression: bool
+    trap: bool
+    mode: Optional[str]
+    event: bool
+    door_ids: List[str]
+    painting_ids: List[str]
+
+    def should_include(self, multiworld: MultiWorld, player: int) -> bool:
+        if self.mode == "colors":
+            return get_option_value(multiworld, player, "shuffle_colors") > 0
+        elif self.mode == "doors":
+            return get_option_value(multiworld, player, "shuffle_doors") > 0
+        elif self.mode == "orange tower":
+            # door shuffle is on and tower isn't progressive
+            return get_option_value(multiworld, player, "shuffle_doors") > 0\
+                and not get_option_value(multiworld, player, "progressive_orange_tower")
+        elif self.mode == "complex door":
+            return get_option_value(multiworld, player, "shuffle_doors") == 2  # complex doors
+        elif self.mode == "door group":
+            return get_option_value(multiworld, player, "shuffle_doors") == 1  # simple doors
+        elif self.mode == "special":
+            return False
+        else:
+            return True
+
+
+class LingoItem(Item):
+    """
+    Item from the game Lingo
+    """
+    game: str = "Lingo"
+
+
+class StaticLingoItems:
+    """
+    Defines the items that can be included in a Lingo world
+    """
+
+    base_id: int = 0
+
+    ALL_ITEM_TABLE: Dict[str, ItemData] = {}
```

Comment:
This is mutable so if you add to this, it gets added for every `StaticLingoItems` in the multiworld. Should definitely not create it here.

Reply:
I'm not the most familiar with Python. What would you recommend I do here? This is intended to be immutable after instantiation.

Reply:
Just instantiating it in the constructor should be fine.

Reply:
Actually it looks like you're trying to use it as if it's a static class, which isn't a thing in Python. You should probably make the table root level and just build it with a base level function that gets called when the file is imported, but it does a check to see if the table has already been built.